src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs

```diff
@@ -7651,10 +7655,11 @@ unsafe class C
     int* p = stackalloc int[1];
 }
 ";
-            CreateCompilation(text, options: TestOptions.UnsafeReleaseDll).VerifyDiagnostics(
-                // (4,14): error CS1525: Invalid expression term 'stackalloc'
+            CreateCompilationWithMscorlibAndSpan(text, options: TestOptions.UnsafeReleaseDll).VerifyDiagnostics(
+                // (4,14): error CS8346: Conversion of a stackalloc expression of type 'int' to type 'int*' is not possible.
```

Comment:
>CS8346: Conversion of a stackalloc expression of type 'int' to type 'int*' is not possible. [](start = 33, length = 91)

The error seems confusing to me since the same conversion is possible for a local declaration. Should this be a specific error that `stackalloc` is not allowed in this context?

Reply:
But it *is* allowed in this context. It just always gives you a `Span<int>`.

Local declarations are treated specially, and even though the looks syntactically the same as a local declaration, it isn't.

---
In reply to: [209301880](https://github.com/dotnet/roslyn/pull/28969#discussion_r209301880) [](ancestors = 209301880)